scripts/validate-requirements.sh

```diff
@@ -0,0 +1,274 @@
+    if [ "$(echo "$END_USER_CREDENTIAL_OUTPUT" | grep -c unset)" -eq 1 ]; then
+        echo "You must configure an End User Credential."
+        echo "Visit https://cloud.google.com/sdk/gcloud/reference/auth/login and follow the instructions to authorize gcloud to access the Cloud Platform with Google user credentials."
+        ERRORS+=$'gcloud end user credential not configured.\n'
+    fi
+
+    APPLICATION_DEFAULT_CREDENTIAL_OUTPUT="$(gcloud auth application-default print-access-token 2>&1 >/dev/null)"
+    if [ "$(echo "$APPLICATION_DEFAULT_CREDENTIAL_OUTPUT" | grep -c 'Could not automatically determine credentials')" -eq 1 ]; then
+        echo "You must configure an Application Default Credential."
+        echo "Visit https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login and follow the instructions to authorize gcloud to access the Cloud Platform with Google user credentials."
+        ERRORS+=$'gcloud application default credential not configured.\n'
+    fi
+}
+
+# Function to validate the Configuration of the Gcloud CLI
+function validate_credential_roles(){
+    check_org_level_roles "$END_USER_CREDENTIAL" "$ORGANIZATION_ID"
+    check_billing_account_roles "$END_USER_CREDENTIAL" "$BILLING_ACCOUNT"
+}
+
+# Verifies wheter a user is assigned to the expected Orgzanization Level roles
+function check_org_level_roles(){
+
+    ORG_LEVEL_ROLES_OUTPUT=$(
+        gcloud organizations get-iam-policy "$2" \
+        --filter="bindings.members:$1" \
+        --flatten="bindings[].members" \
+        --format="table(bindings.role)" 2>/dev/null)
+
+    lines=$(echo "$ORG_LEVEL_ROLES_OUTPUT" | grep -o -e roles/resourcemanager.folderCreator -e roles/resourcemanager.organizationAdmin -e roles/orgpolicy.policyAdmin | wc -l)
+
+    if [ "$lines" -ne 3 ]; then
+        echo "The User must have the Organization Roles resourcemanager.folderCreator, resourcemanager.organizationAdmin and roles/orgpolicy.policyAdmin"
+        ERRORS+=$'There are missing organization level roles on the Credential.\n'
+    fi
+}
+
+# Verifies wheter a user is assigned to the expected Billing Level roles
+function check_billing_account_roles(){
+
+    BILLING_LEVEL_ROLES_OUTPUT=$(
+        gcloud beta billing accounts get-iam-policy "$2" \
+        --filter="bindings.members:$1" \
+        --flatten="bindings[].members" \
+        --format="table(bindings.role)" 2>/dev/null)
+
+    lines=$(echo "$BILLING_LEVEL_ROLES_OUTPUT" | grep -o -e roles/billing.admin | wc -l)
+
+    if [ "$lines" -ne 1 ]; then
+        echo "The User must have the Billing Account Role billing.admin"
+        ERRORS+=$'There are missing billing account level roles on the Credential.\n'
+    fi
+}
+
+# Checks if initial config was done for 0-bootstrap step
+function validate_bootstrap_step(){
+    FILE=0-bootstrap/terraform.tfvars
+    if [ ! -f "$FILE" ]; then
+        echo "$FILE does not exists."
```

Comment:
Shouldn't this as the user to also rename the .example.tfvars to .tfvars?

Reply:
@bharathkkb sorry but I couldn't get the idea, could you give a more descriptive comment about the issue?